ci/ci/ci.py

```diff
@@ -558,6 +569,75 @@ async def unfreeze_deploys(request, userdata):  # pylint: disable=unused-argumen
     return web.HTTPFound(deploy_config.external_url('ci', '/'))
 
 
+@routes.get('/namespaces')
+@web_authenticated_developers_only()
+async def get_active_namespaces(request, userdata):
+    db: Database = request.app['db']
+    namespaces = [
+        r
+        async for r in db.execute_and_fetchall(
+            '''
+SELECT active_namespaces.*, JSON_ARRAYAGG(service) as services
+FROM active_namespaces
+LEFT JOIN deployed_services
+ON active_namespaces.namespace = deployed_services.namespace
+GROUP BY active_namespaces.namespace'''
+        )
+    ]
+    for ns in namespaces:
+        ns['services'] = [s for s in json.loads(ns['services']) if s is not None]
+    context = {
+        'namespaces': namespaces,
+    }
+    return await render_template('ci', request, userdata, 'namespaces.html', context)
+
+
+@routes.post('/namespaces/{namespace}/services/add')
+@check_csrf_token
+@web_authenticated_developers_only()
+async def add_namespaced_service(request, userdata):  # pylint: disable=unused-argument
+    db: Database = request.app['db']
+    post = await request.post()
+    service = post['service']
+    namespace = request.match_info['namespace']
+
+    await db.execute_insertone(
+        'INSERT INTO deployed_services VALUES (%s, %s)',
```

Comment:
I think you need some sort of check here or nice error if the namespace/service already exists.